api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"put": {
+				"summary": "Create or update an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "endpoint",
+						"in": "body",
+						"required": true,
+						"schema": {
+							"$ref": "#/definitions/Endpoint"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "Success",
+						"schema": {
+							"$ref": "#/definitions/EndpointResponse"
+						}
+					},
+                                        "405": {
+                                                "description": "Endpoint already exists",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        },
+                                        "default": {
+                                                "description": "Unexpected error",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"delete": {
+				"description": "Delete an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			}
+		},
+                "/endpoint/list": {
+			"get": {
+				"summary": "Retrieve list of all endpoints",
+				"tags": [ "endpoint" ],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+							"type": "array",
+						        "items": {
+								"$ref": "#/definitions/Endpoint"
+							}
+                                                }
+					}
+				}
+			}
```

Comment:
We can add support for this via `PUT /endpoint/batch/` which assumes an array of endpoints and returns an array of return codes. I think it makes sense to keep this separate from the existing operation as it will require a policy argument of sort to define whether to continue creating up error.